core/src/main/scala/org/apache/spark/deploy/security/HadoopDelegationTokenManager.scala

```diff
@@ -42,7 +42,7 @@ import org.apache.spark.internal.Logging
 private[spark] class HadoopDelegationTokenManager(
     sparkConf: SparkConf,
     hadoopConf: Configuration,
-    fileSystems: Set[FileSystem])
+    fileSystems: () => Set[FileSystem])
```

Comment:
I think this closure needs to take a `Configuration` as paramater.

If you look at the old code, when renewing tokens, the file system list uses `AMCredentialRenewer.freshHadoopConf` as the configuration, not the `ApplicationMaster` configuration as your code is doing. That sounds more correct.